src/irmin-pack/unix/async.ml

```diff
@@ -108,10 +108,10 @@ module Unix = struct
   let await t =
     match t.status with
     | `Running ->
-        let+ pid, status = Lwt_unix.waitpid [] t.pid in
+        let pid, status = Unix.waitpid [] t.pid in
```

Comment:
I'm a bit concerned about this change - `Lwt_unix.waitpid` blocked the current promise, but allowed for other Lwt promises to run. I think with this change we are blocking all promises in this thread?

Reply:
Hmm, this is a good question. We do not use `Lwt` for other I/O operations (like in `io.ml`) so I'm wondering if this change makes anything that much worse. In the high-level gc api (that is used by Tezos), `await` will never be called -- we only check `status`. Does that change anything for you?

Reply:
I agree we should avoid doing this as it is blocking the whole process that defeat the purpose of that API

Reply:
@samoht this function is not called as a part of the high-level api. It only calls `status`, which is already using `Unix.waitpid [ Unix.WNOHANG ] t.pid`.